### ISSUE TITLE
Colorful file icons

### ISSUE THREAD

**Issue by probablykasper** [zed-industries/zed#4609]
### Check for existing issues

- [X] Completed

### Describe the feature

File icons were added (yay!) but I struggle to differentiate the monochrome icons
<img width="235" alt="image" src="https://github.com/zed-industries/community/assets/11315492/ec059305-6520-45d3-a4a3-64cc8208bc5c">

Adding colors would help me a lot
<img width="252" alt="image" src="https://github.com/zed-industries/community/assets/11315492/288817fe-eafb-49b1-a615-013a8b12cd53">

Related:
- https://github.com/zed-industries/zed/issues/6150
- https://github.com/zed-industries/zed/issues/5525

**Comment by Ymirke** [comment]
Second this!

Suggest icon themes for folders as well. 

The left image is a lot faster to navigate, as it's simpler to differentiate between folders.

<img width="299" alt="Screenshot" src="https://github.com/zed-industries/zed/assets/36711026/c92f35ca-9ba1-4906-abd1-93471cab33d5">

<img width="415" alt="Screenshot 2024-01-26 at 23 00 19" src="https://github.com/zed-industries/zed/assets/36711026/75ce0ba1-358e-480d-8bda-738c77596452">

**Comment by nguyenthanhan201** [comment]
I hope this feature released soon

**Comment by yayashn** [comment]
Lack of colour coded icons actually makes it so hard to navigate a codebase, especially a new one.

**Comment by ElectricS01** [comment]
This would be helpful

**Comment by maxdeviant** [comment]
I think this would fall under #8843.

**Comment by Moshyfawn** [comment]
> I think this would fall under #8843.

I'm going to close this in favour of #8843, as it makes sense for the community to contribute their own icons, rather than for the Zed team to maintain multiple sets.

Be sure to +👍 the extension issue.

**Comment by probablykasper** [comment]
@Moshyfawn Was it considered to make the current icons colorful instead?

**Comment by maxdeviant** [comment]
> @Moshyfawn Was it considered to make the current icons colorful instead?

The current icon style is an intentional design decision, so I don't see us changing it.

**Comment by LL-Program** [comment]
> I hope this feature released soon

You should check out my own code Editor I will relasing it soon with more feautures and fully customizable UI and faster speed

**Comment by SealinGp** [comment]
Looking forward to release this feature !

**Comment by 0xFlo** [comment]
so you decided to keep them monochrom? How do I make them colorful as suggested by OP?

**Comment by othavioquiliao** [comment]
I would also like to know, it's the only thing that's holding me back from switching 100% to Zed. 
And it's probably a problem for new users too, as it makes readability worse and slower.

**Comment by yayashn** [comment]
> I would also like to know, it's the only thing that's holding me back from switching 100% to Zed. And it's probably a problem for new users too, as it makes readability worse and slower.

> Lack of colour coded icons actually makes it so hard to navigate a codebase, especially a new one.

So surprisingly even though I was a pro-coloured-files guy not too long ago, a moment of rage in VSCode persuaded me to stick with Zed despite the lack of colour.

Was painfully unproductive initially - but after using Zed I was forced to discover that navigating using the file explorer should not be the primary method of navigation. So what should the primary method be? Well I've found these 2 techniques to work best for me & I suggest people who are overly-reliant on the file explorer to give them a shot:

**Method 1: cmd + p **  (VSCode bindings)
Know the file name or at least some of it? cmd + p to open up the quick search tool, type the first few letters, and there's your file. 

**Method 2: Workspace Search**
Don't know the name of the file but know some of the code inside it? Use workspace search and enter some code relatively unique to the file to immediately find the file. It's not keybound by default but to test it out before setting up a keybind you can do `cmd + shift + p` -> `search for workspace search` -> `enter`.

Perhaps this is an obvious work flow for most, but I suspect people like me who are heavily advocating for file colours are not using this. Also if anyone knows any better ways of navigating the project feel free to point them out! 

**tldr**
The demand I had for file colours was merely a symptom of over-reliance on the explorer, and I suspect this is the case for other people as well so I suggest they explore the imo superior ways of navigating projects.

**Comment by 0xFlo** [comment]
@yayashn, you raise some valid points, but honestly, Zed is way too opinionated for my taste. I wasn’t expecting Neovim-level customization, but at least give users the option to decide whether they want to stick with millennial grey or add some actual color. Even Apple lets you customize your homescreen these days 😵

**Comment by yayashn** [comment]
@0xFlo Yeah it's a valid feature request regardless.

**Comment by jere-co** [comment]
@yayashn, I don't think the the color icons and `cmd + p` solve the same problem. I'm also navigating around the project without looking at the file tree, but when jumping to a new, or going back to an old project, it's mandatory to look around what files there are. I'm still opening such projects in VSCode, because it's just that much better experience with colored icons.

**Comment by singularitti** [comment]
It's closed in favor of #8843 @iqmal852.
